### ISSUE TITLE
package not debuggable error

### ISSUE THREAD

**Issue by Talk2Giuseppe** [89jd/hass-roborock#5]
I get the "run-as: package not debuggable: com-roborock.smart" error when attempting to issue the following command:

adb exec-out run-as com.roborock.smart find /data/data/com.roborock.smart/files/rr_cache/ | egrep rr_tuya_[0-9] | xargs -n1 adb exec-out run-as com.roborock.smart cat > roborock_details.gz 
gzip -d roborock_details.gz 

Actually, since the command error'd out I parsed the commands to figure out which one was causing the issue.  It was the first command: "adb exec-out run-as com.roborock.smart find /data/data/com.roborock.smart/files/rr_cache/" that produced the error.

Anyone have any experience with overcoming this obstacle?

Thanks!

**Comment by 89jd** [comment]
Are you running on rooted device?

**Comment by Talk2Giuseppe** [comment]
Yes, the device is rooted.  It's Note 8 running HadesROM.  I am able to access the folder from where the files are on the device itself.  But when running the adb commands from a terminal window withing linux, I get those errors.  I can also confirm that when I connect the phone to the linux machine, it does prompt/confirm that I want to be in debug mode.  

Any ideas?

**Comment by Talk2Giuseppe** [comment]
What version of the roborock software do you have installed?  I am beginning to think that perhaps this current version is signed differently than perhaps an earlier version of the apk which is what is producing the error.

**Comment by sShrek1** [comment]
Just got the same issue. App Version 2.3.30, Jan 14. 
After running first command I get an roborock_details.gz. roborock_details.gz is an ASCII text type.

**Comment by bbr111** [comment]
Same here. Testet different Versions on different Emus ( Nox, Bluestacks) with root or without.
Tried with original apk and apk from https://github.com/redphx/roborock-tuya-token

roborock-tuya-token apk is currently not running if account is Region EU.

**Comment by grinco** [comment]
Looks like you need debugging enabled in the app for this to work. Tried backing up settings using ADB - won't work either. I'm trying it on a non-rooted device, with no luck. on a rooted one, I'd just dump the cache directory. Does anyone know where to get a debug package? I'm also thinking of installing the app in a bluestacks or any other emulator and try getting the root there.

**Comment by RubenKelevra** [comment]
Hey guys,

as long as your phone is rooted, that's pretty easy to solve.

- Install an android terminal
- open it
- type `su`
- you'll be asked to give it superuser rights
- type `zcat /data/data/com.roborock.smart/files/rr_cache/rr*`

This should post all info on your screen you need, you just have to search for the `localKey`.

**Comment by sShrek1** [comment]
> Hey guys,
> 
> as long as your phone is rooted, that's pretty easy to solve.
> 
> - Install an android terminal
> - open it
> - type `su`
> - you'll be asked to give it superuser rights
> - type `zcat /data/data/com.roborock.smart/files/rr_cache/rr*`
> 
> This should post all info on your screen you need, you just have to search for the `localKey`.

Which device and android version did you test this on? 
I can't get zcat to work, running android 6.0
